emails/views.py

```diff
@@ -581,10 +581,21 @@ def _get_keys_from_headers(headers):
         if header["name"].lower() == "in-reply-to":
             in_reply_to = header["value"]
             message_id_bytes = get_message_id_bytes(in_reply_to)
-    if in_reply_to is None:
-        incr_if_enabled("mail_to_replies_without_in_reply_to", 1)
-        raise InReplyToNotFound
-    return derive_reply_keys(message_id_bytes)
+            return derive_reply_keys(message_id_bytes)
+
+        if header["name"].lower() == "references":
+            message_ids = header["value"]
+            for message_id in message_ids.split(" "):
+                message_id_bytes = get_message_id_bytes(message_id)
+                lookup_key, encryption_key = derive_reply_keys(message_id_bytes)
+                try:
+                    # FIXME: calling code is likely to duplicate this query
+                    _get_reply_record_from_lookup_key(lookup_key)
+                    return lookup_key, encryption_key
+                except Reply.DoesNotExist:
+                    pass
+            raise Reply.DoesNotExist
+    raise ReplyHeadersNotFound
```

Comment:
Should we still call `incr_if_enabled("mail_to_replies_without_in_reply_to", 1)`?

Reply:
Good call. I changed it to `mail_to_replies_without_reply_headers`.